### PR TITLE
feat(snap): add additional tokens for app-service-configurable profiles

### PR DESCRIPTION
**Pull request by siggiskulason**
This PR adds updates the snap so that the secret-store tokens for
1. external-mqtt-trigger
2. push-to-core

are copied from the edgexfoundry snap. Both are built-in profiles that are referenced in the TAF tests

Signed-off-by: Siggi Skulason <siggi.skulason@canonical.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-service-configurable/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) N/A
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) N/A
  <link to docs PR>

## Testing Instructions

To test:
1.  This PR depends on the edgexfoundry also being updated in PR https://github.com/edgexfoundry/edgex-go/pull/3825. Use the edgexfoundry snap from that PR. 
2. Build the edgex-app-service-configurable snap
3. Install it and then connect the interfaces so that it gets the token: `sudo snap connect edgexfoundry:edgex-secretstore-token edgex-app-service-configurable`
4. set the profile:`sudo snap set edgex-app-service-configurable profile=external-mqtt-trigger`
5. start the service and confirm in the journal that it has started with the secret-store token

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ slots:
     interface: content
     content: edgex-secretstore-token
     source:
-      write: [$SNAP_DATA/app-http-export, $SNAP_DATA/app-mqtt-export, $SNAP_DATA/app-functional-tests]
+      write: [$SNAP_DATA/app-http-export, $SNAP_DATA/app-mqtt-export, $SNAP_DATA/app-external-mqtt-trigger, $SNAP_DATA/app-functional-tests]
```
[the remark this answers is not in the document]
Comment (farshidtz) [reply snap/snapcraft.yaml:38]: Assuming the last two are only needed for tests. Change the comment to what makes sense.

[the remark this answers is not in the document]
Comment (siggiskulason) [reply snap/snapcraft.yaml]: I just pushed a 2nd commit to
1. change this to a list
2. add the push-to-core profile as well.

**Review by farshidtz — APPROVED** [review]
Looks good.

But please update the PR title:
- multiple tokens added now
- fix or feat?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -35,7 +35,13 @@ slots:
...
     content: edgex-secretstore-token
     source:
-      write: [$SNAP_DATA/app-http-export, $SNAP_DATA/app-mqtt-export, $SNAP_DATA/app-functional-tests]
+      write: 
+        - $SNAP_DATA/app-http-export
+        - $SNAP_DATA/app-mqtt-export
+        - $SNAP_DATA/app-external-mqtt-trigger
+        - $SNAP_DATA/app-functional-tests
+        - $SNAP_DATA/app-push-to-core
+        - $SNAP_DATA/app-rules-engine
```
Comment (farshidtz) [inline snap/snapcraft.yaml:44]: In future, we need to improve the way this content interface works to allow receiving tokens that are requested dynamically from the secret store, e.g. for development, testing, multiple custom names.

**Comment by siggiskulason** [comment]
> Looks good.
> 
> But please update the PR title:
> 
> * multiple tokens added now
> * fix or feat?

Done

**Pull request merged**